### PR TITLE
Add Header and Cleanup

### PR DESCRIPTION
**Pull request by cchighman**
@shivsood 
This PR has the header introduced at the front of each file along with cleanup.  90% of the cleanup is apply Scala linting, styling, etc.  10% of the cleanup was very minor refactoring.  It made life much easier to include that work in this PR.  I can add very detailed comments to identify changes.

A majority of the refactoring was actually around splitting up or adding new tests.  I did not include the new or adjusted tests in this PR.

### CODE REVIEW

**Comment by cchighman** [comment]
I may re-open this PR so my commits list is cleaned up a bit better.

File: license.txt
Comment (arvindshmicrosoft) [inline license.txt:2]: AFAIK this project is Apache 2.0 license, so we would need to modify this accordingly.

**Comment by shivsood** [comment]
Thanks @cchighman . Looks great to me. 'll mark this for a test run on BDC; but i dont expect any issue. 
@arvindshmicrosoft  @rajmera3 will need your approval for header licenses  etc.

**Comment by arvindshmicrosoft** [comment]
> Thanks @cchighman . Looks great to me. 'll mark this for a test run on BDC; but i dont expect any issue.
> @arvindshmicrosoft @rajmera3 will need your approval for header licenses etc.

Shiv - the headers mismatch the license - the Apache license is differently worded. I've requested changes from Christopher accordingly...

**Comment by shivsood** [comment]
> ange

Thanks Arvind.

**Comment by shivsood** [comment]
Needs @arvindshmicrosoft and @rajmera3 review for license texts.

**Comment by arvindshmicrosoft** [comment]
> Needs @arvindshmicrosoft and @rajmera3 review for license texts.

@shivsood I have already requested changes to use the Apache license instead of the MIT license. Let's wait for @cchighman

**Comment by cchighman** [comment]
Sorry fro delay here

**Comment by arvindshmicrosoft** [comment]
@cchighman sorry to ping again - would you be able to address the changes needed?

**Comment by cchighman** [comment]
> @cchighman sorry to ping again - would you be able to address the changes needed?

Yes.  Ill handle this today.

**Comment by cchighman** [comment]
> > @cchighman sorry to ping again - would you be able to address the changes needed?
> 
> Yes. Ill handle this today.

This evening :)

**Comment by cchighman** [comment]
This is on my agenda this morning

File: src/main/scala/com/microsoft/sqlserver/jdbc/spark/failureinjection/FailureInjection.scala
Comment (arvindshmicrosoft) [inline src/main/scala/com/microsoft/sqlserver/jdbc/spark/failureinjection/FailureInjection.scala:2]: @rajmera3 usually most ASL based projects do not include a copyright line, it is usually just the boilerplate header. Any comments?

Comment (arvindshmicrosoft) [reply src/main/scala/com/microsoft/sqlserver/jdbc/spark/failureinjection/FailureInjection.scala:2]: @rajmera3 - can you clarify my question above?

Comment (rajmera3) [reply src/main/scala/com/microsoft/sqlserver/jdbc/spark/failureinjection/FailureInjection.scala:2]: @arvindshmicrosoft You're right - we should remove the MS copyright headers

Comment (arvindshmicrosoft) [reply src/main/scala/com/microsoft/sqlserver/jdbc/spark/failureinjection/FailureInjection.scala:2]: @cchighman - sorry, could you please remove the copyright headers as per this comment thread? Thanks again for your follow-up.

Comment (cchighman) [reply src/main/scala/com/microsoft/sqlserver/jdbc/spark/failureinjection/FailureInjection.scala:2]: Yes

**Review by arvindshmicrosoft — APPROVED** [review]
@rajmera3 and @shivsood please take a final look. The changes LGTM.

**Review by rajmera3 — APPROVED** [review]

**Comment by rajmera3** [comment]
Looks good to me! Will leave to Shiv for the final ack and merge @shivsood

**Comment by shivsood** [comment]
Will need a round of test before the final merge to ensure no regressions/issues. 'll validate before merge.

**Comment by shivsood** [comment]
@cchighman the branch needs to be baseline from master resolve conflicts. Can u please take care of that?

Also a data pool test failed with this branch. Need to look into this. @luxu1-ms 

def test_gci_dp_read_write() {
        val table_name = s"test_data_pool_write1"
        val df = testUtils.create_toy_df()
        testUtils.df_write(df, SaveMode.Overwrite, table_name, dataPoolDataSource = testUtils.dataPoolExternalDataSource)

        // Insert into the external tables ( Append mode)
        testUtils.df_write(df, SaveMode.Append, table_name, dataPoolDataSource = testUtils.dataPoolExternalDataSource)
        val new_df = testUtils.df_read(table_name, dataPoolDataSource = testUtils.dataPoolExternalDataSource)
        assert(2*df.count == new_df.count)
        testUtils.drop_external_table(table_name)
    }

```
2020-10-24 01:19:12,089 DEBUG hdfs.DataStreamer: blk_1073746979_6157 waiting for ack for: 25
2020-10-24 01:19:12,089 DEBUG hdfs.DataStreamer: stage=DATA_STREAMING, blk_1073746979_6157
2020-10-24 01:19:12,093 DEBUG hdfs.DataStreamer: blk_1073746979_6157 sending packet seqno: 25 offsetInBlock: 154624 lastPacketInBlock: false lastByteOffsetInBlock: 155198
2020-10-24 01:19:12,096 ERROR yarn.ApplicationMaster: User class threw exception: java.lang.reflect.InvocationTargetException
...

Caused by: java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:156)
	at DataPoolTests.test_gci_dp_read_write(DataPoolTest.scala:57)
	... 25 more

```

**Comment by cchighman** [comment]
> @cchighman the branch needs to be baseline from master resolve conflicts. Can u please take care of that?
> 
> Also a data pool test failed with this branch. Need to look into this. @luxu1-ms
> 
> def test_gci_dp_read_write() {
> val table_name = s"test_data_pool_write1"
> val df = testUtils.create_toy_df()
> testUtils.df_write(df, SaveMode.Overwrite, table_name, dataPoolDataSource = testUtils.dataPoolExternalDataSource)
> 
> ```
>     // Insert into the external tables ( Append mode)
>     testUtils.df_write(df, SaveMode.Append, table_name, dataPoolDataSource = testUtils.dataPoolExternalDataSource)
>     val new_df = testUtils.df_read(table_name, dataPoolDataSource = testUtils.dataPoolExternalDataSource)
>     assert(2*df.count == new_df.count)
>     testUtils.drop_external_table(table_name)
> }
> ```
> 
> ```
> 2020-10-24 01:19:12,089 DEBUG hdfs.DataStreamer: blk_1073746979_6157 waiting for ack for: 25
> 2020-10-24 01:19:12,089 DEBUG hdfs.DataStreamer: stage=DATA_STREAMING, blk_1073746979_6157
> 2020-10-24 01:19:12,093 DEBUG hdfs.DataStreamer: blk_1073746979_6157 sending packet seqno: 25 offsetInBlock: 154624 lastPacketInBlock: false lastByteOffsetInBlock: 155198
> 2020-10-24 01:19:12,096 ERROR yarn.ApplicationMaster: User class threw exception: java.lang.reflect.InvocationTargetException
> ...
> 
> Caused by: java.lang.AssertionError: assertion failed
> 	at scala.Predef$.assert(Predef.scala:156)
> 	at DataPoolTests.test_gci_dp_read_write(DataPoolTest.scala:57)
> 	... 25 more
> ```

Yes, my apologies for the delays.  I've been out of office and in process of moving into a home we purchased inbetween Redmond and Fall City.  I have an open window to do this now.

**Comment by shivsood** [comment]
Passed all test on Spark2.4 on BDC.

**Pull request merged**
